dpnp/dpnp_array.py

```diff
@@ -585,7 +585,19 @@ def flatten(self, order='C'):
         return new_arr
 
  # 'getfield',
- # 'imag',
+
+    @property
+    def imag(self):
+        """
+        Return the imaginary part of the complex argument.
+
+        """
+
+        return dpnp.imag(self)
+
+    @imag.setter
+    def imag(self, value):
+        dpnp.imag(self)[:] = value
```

Comment:
This setter only makes sense when `dpnp.imag` is a view into original array's allocation. This is clearly not true for arrays with non-complex types.